### PR TITLE
Miscellaneous Fixes

### PR DESCRIPTION
**Pull request by squarebracket**
I am basically just repeating the git commit logs, but here we go:
- Ensure log directory, make OS-agnostic
- Fix old parameter usage (I believe you already fixed this)
- Update Apache's `graphite.conf` template to point to `graphite_wsgi.py`
- Update README to reflect this
- Ensure `graphiteweb_storage_dir_REAL` directory
- Remove old `wsgi.py` link for gunicorn (no longer needed)
- Notify web server on changes to `local_settings.py` and `graphite_wsgi.py`
- Properly parameterize `--chdir` in gunicorn `init.d` file
- Add missing `service_provider` for debian
- Add case for `$::service_provider == 'systemd'` in `config.pp`

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -47,7 +47,8 @@
       $apache_wsgi_socket_prefix = '/var/run/apache2/wsgi'
       $apacheconf_dir            = '/etc/apache2/sites-available'
       $apacheports_file          = 'ports.conf'
-      $service_provider          = undef
+      $apache_logdir_graphite    = '/var/log/apache2/graphite-web'
+      $service_provider          = $::service_provider
```
Comment (dwerder) [inline manifests/params.pp]: All CI tests for Debian and Ubuntu are failing, because of the unknown variable $::service_provider.

**Comment by dwerder** [comment]
The fact `service_provider` is first introduced in stdlib 4.10.0, which is pretty new. At the moment the module requirements for stdlib are 4.6. So this fact is not know and can not be expected to exist. If we want to use it we have to bump the stdlib version.

**Comment by squarebracket** [comment]
Ah, ok. I'll push a fix.

**Comment by squarebracket** [comment]
This isn't an enhancement, it's fixing several bugs.

**Comment by dwerder** [comment]
There are no bugs... just hidden features ;-)

**Comment by dwerder** [comment]
I applied your fixes without the service_provider fix.

**Pull request closed without merging**